src/Core/Domain/CatalogPriceRule/Command/AddCatalogPriceRuleCommand.php

```diff
@@ -0,0 +1,259 @@
+     */
+    public function __construct(
+        string $name,
+        int $currencyId,
+        int $countryId,
+        int $groupId,
+        int $fromQuantity,
+        string $reductionType,
+        float $reductionValue,
+        int $shopId,
+        bool $includeTax,
+        float $price
+    ) {
+        $this->name = $name;
+        $this->currencyId = $currencyId;
+        $this->countryId = $countryId;
+        $this->groupId = $groupId;
+        $this->fromQuantity = $fromQuantity;
+        $this->reduction = new Reduction($reductionType, $reductionValue);
+        $this->shopId = $shopId;
+        $this->price = $price;
+        $this->includeTax = $includeTax;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return int
+     */
+    public function getShopId(): int
+    {
+        return $this->shopId;
+    }
+
+    /**
+     * @return int
+     */
+    public function getCurrencyId(): int
+    {
+        return $this->currencyId;
+    }
+
+    /**
+     * @return int
+     */
+    public function getCountryId(): int
+    {
+        return $this->countryId;
+    }
+
+    /**
+     * @return int
+     */
+    public function getGroupId(): int
+    {
+        return $this->groupId;
+    }
+
+    /**
+     * @return int
+     */
+    public function getFromQuantity(): int
+    {
+        return $this->fromQuantity;
+    }
+
+    /**
+     * @return Reduction
+     */
+    public function getReduction(): Reduction
+    {
+        return $this->reduction;
+    }
+
+    /**
+     * @return float
+     */
+    public function getPrice(): float
+    {
+        return $this->price;
+    }
+
+    /**
+     * @return DateTime|null
+     */
+    public function getDateTimeFrom(): ?DateTime
+    {
+        return $this->dateTimeFrom;
+    }
+
+    /**
+     * @return DateTime|null
+     */
+    public function getDateTimeTo(): ?DateTime
+    {
+        return $this->dateTimeTo;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isTaxIncluded(): bool
+    {
+        return $this->includeTax;
+    }
+
+    /**
+     * @param string $dateTimeFrom
+     *
+     * @throws CatalogPriceRuleConstraintException
+     */
+    public function setDateTimeFrom(string $dateTimeFrom)
+    {
+        $this->dateTimeFrom = $this->createDateTime($dateTimeFrom);
+    }
+
+    /**
+     * @param string $dateTimeTo
+     *
+     * @throws CatalogPriceRuleConstraintException
+     */
+    public function setDateTimeTo(string $dateTimeTo)
+    {
+        $this->dateTimeTo = $this->createDateTime($dateTimeTo);
+    }
+
+    /**
+     * @param string $dateTime
+     *
+     * @return DateTime
+     *
+     * @throws CatalogPriceRuleConstraintException
+     */
+    private function createDateTime($dateTime): DateTime
+    {
+        try {
+            return new DateTime($dateTime);
+        } catch (Exception $e) {
```

Comment:
Are you sure every Exception thrown from previous line are because of bad format 😄?

Reply:
As the phpdoc says, yes.
> If time contains an invalid date/time format, then an exception is now thrown. Previously an error was emitted

Reply:
indeed, exception should be thrown only due to wrong format :astonished: . Imo, the current message looks clear anyway